### PR TITLE
[#68] Integrate with real data on the surveys list & pagination

### LINKED ISSUE #68: [Integrate] As a logged in user, I can see the surveys on the Home screen
## Why

Perform the integration for the pagination data of surveys and binding to the UI.
 
## Acceptance Criteria
- Logged in users must be allowed to see the real surveys list
- When they scroll to nearly the end, fetch the next batch of data
- Keep the indicator list stat updated correctly.

### PR DESCRIPTION
**Pull request by sleepylee**
Closes #68 

## What happened 👀

Binding the data to the list, display the Surveys info (not yet for the list indicator)
 
## Insight 📝

- Bind the data for the Surveys; exposing data from Controller (ViewModel) to the View.
- Add a listener to update/fetch more when the users landing on nearly the end of the current survey's list.
- Adding indicator updates.
 
## Proof Of Work 📹
All the surveys can be fetched now until the end:

https://user-images.githubusercontent.com/13435717/123775924-0064e280-d8f9-11eb-8e45-88f408b145cb.mov

### CODE REVIEW

File: lib/pages/home/home_controller.dart [hunk 1]
```diff
@@ -39,14 +56,41 @@ class HomeController extends GetxController {
...
-      surveys.value.forEach((element) {
-        print("Items: ${element.title} - ${element.description}");
-      });
+    // ignore: invalid_use_of_protected_member
+    if (_surveys.value.isEmpty) {
+      final surveys = await getSurveysUseCase.call("");
+      if (surveys is Success<List<Survey>>) {
+        surveys.value.forEach((element) {
+          _surveys.add(element);
+        });
```
Comment (manh-t) [inline lib/pages/home/home_controller.dart:71]: can we use `addAll` instead? 🙏 
```suggestion
        _surveys.addAll(surveys.value);
```

File: lib/pages/home/home_controller.dart [hunk 2]
```diff
@@ -39,14 +56,41 @@ class HomeController extends GetxController {
...
+      final surveys = await getSurveysUseCase.call(lastCursor);
+      if (surveys is Success<List<Survey>>) {
+        if (surveys.value.isEmpty) {
+          _paginationFinished = true;
+          return;
+        }
+
+        surveys.value.forEach((element) {
+          _surveys.add(element);
+        });
```
Comment (manh-t) [inline lib/pages/home/home_controller.dart:85]: same here

**Review by manh-t — COMMENTED** [review]
just an improvement, rest LGTM 🙏

**Comment by sleepylee** [comment]
all updated @manh-t 🙇  added a small code part for the indicator as well;

**Review by manh-t — APPROVED** [review]

**Review by markgravity — APPROVED** [review]
LGTM 🤖

**Pull request merged**